src/CMakeLists.txt

```diff
@@ -481,7 +478,7 @@ endif(${HAVE_ARMV8_CRC})
 
 add_library(common_utf8 STATIC common/utf8.c)
 
-target_link_libraries(common json_spirit common_utf8 erasure_code rt uuid ${CRYPTO_LIBS} ${Boost_LIBRARIES} ${BLKID_LIBRARIES} ${EXECINFO_LIBRARIES})
```

Comment:
@alimaredia it is removed in fa4934c.  test_build_libcommon fails to link with it

```
[100%] Linking CXX executable ../../bin/test_build_libcommon
../../lib/liberasure_code.so: undefined reference to `cpp_strerror[abi:cxx11](int)'
../../lib/liberasure_code.so: undefined reference to `Mutex::Lock(bool)'
../../lib/liberasure_code.so: undefined reference to `Mutex::Unlock()'
../../lib/liberasure_code.so: undefined reference to `get_str_list(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::list<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >&)'
../../lib/liberasure_code.so: undefined reference to `Mutex::Mutex(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool, bool, CephContext*)'
../../lib/liberasure_code.so: undefined reference to `Mutex::~Mutex()'
collect2: error: ld returned 1 exit status
src/test/CMakeFiles/test_build_libcommon.dir/build.make:124: recipe for target 'bin/test_build_libcommon' failed
make[3]: *** [bin/test_build_libcommon] Error 1
CMakeFiles/Makefile2:7840: recipe for target 'src/test/CMakeFiles/test_build_libcommon.dir/all' failed
make[2]: *** [src/test/CMakeFiles/test_build_libcommon.dir/all] Error 2
CMakeFiles/Makefile2:7852: recipe for target 'src/test/CMakeFiles/test_build_libcommon.dir/rule' failed
make[1]: *** [src/test/CMakeFiles/test_build_libcommon.dir/rule] Error 2
Makefile:2383: recipe for target 'test_build_libcommon' failed
make: *** [test_build_libcommon] Error 2
```

so i think we should remove it. since libcommon is just an internal library. and the purpose of test_build_libcommon is to make sure we can link against it without referencing g_ceph_context. and i don't think it makes sense include erasure_code in libcommon? if you have concern with it i can bring it back. and mark erasure_code as a STATIC library.